File: backend/internal/models/book.go
package models

import "time"

type Book struct {
	ID             uint      `gorm:"primaryKey" json:"id"`
	Title          string    `gorm:"size:255;not null;index" json:"title"`
	Authors        string    `gorm:"size:1000;not null;default:'';index" json:"authors"`
	Description    string    `gorm:"type:text" json:"description"`
	Publisher      string    `gorm:"size:255" json:"publisher"`
	PublishedYear  *int      `json:"publishedYear,omitempty"`
	PageCount      *int      `json:"pageCount,omitempty"`
	StoredFilename string    `gorm:"size:255;not null;uniqueIndex" json:"-"`
	OriginalName   string    `gorm:"size:255;not null" json:"originalName"`
	CoverFilename  string    `gorm:"size:255" json:"-"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

File: mobile/lib/core/models/book.dart
class Book {
  const Book({
    required this.id,
    required this.title,
    required this.authors,
    required this.originalName,
    this.description,
    this.publisher,
    this.publishedYear,
    this.pageCount,
  });

  final int id;
  final String title;
  final String authors;
  final String originalName;
  final String? description;
  final String? publisher;
  final int? publishedYear;
  final int? pageCount;

  factory Book.fromJson(Map<String, dynamic> json) => Book(
    id: json['id'] as int,
    title: json['title'] as String,
    authors: json['authors'] as String,
    originalName: json['originalName'] as String? ?? '',
    description: json['description'] as String?,
    publisher: json['publisher'] as String?,
    publishedYear: json['publishedYear'] as int?,
    pageCount: json['pageCount'] as int?,
  );
}
